spec/models/miq_request_spec.rb

```diff
@@ -418,4 +418,34 @@
       expect(request.get_user.current_group).to eq(group1)
     end
   end
+
+  context "#update_request" do
+    before do
+      allow(MiqServer).to receive(:my_zone).and_return("New York")
+    end
+
+    let(:request) do
+      FactoryGirl.create(:miq_provision_request,
+                         :requester => fred,
+                         :options   => {:a => "1"})
+    end
+
+    it "user_message" do
+      msg = "Yabba Dabba Doo"
+      request.update_request({:user_message => msg}, fred)
+
+      request.reload
+      expect(request.options[:user_message]).to eq(msg)
+      expect(request.message).to eq(msg)
+    end
+
+    it "truncates long messages" do
+      msg = "Yabba Dabba Doo" * 30
+      request.update_request({:user_message => msg}, fred)
+
+      request.reload
+      expect(request.options[:user_message].length).to eq(255)
+      expect(request.message.length).to eq(255)
+    end
+  end
```

Comment:
We should included in the existing tests that `after_update_options` is not called.  And include one test that passes additional keys and expect that `after_update_options` is called.